--- upstream/vllm/engine/arg_utils.py
+++ installed/vllm/engine/arg_utils.py
@@ -1912,9 +1912,12 @@
                 self.data_parallel_size
                 * self.pipeline_parallel_size
                 * self.tensor_parallel_size
+                * self.prefill_context_parallel_size
             )
             world_size_within_dp = (
-                self.pipeline_parallel_size * self.tensor_parallel_size
+                self.pipeline_parallel_size
+                * self.tensor_parallel_size
+                * self.prefill_context_parallel_size
             )
             local_world_size = world_size // self.nnodes
             assert world_size % self.nnodes == 0, (
